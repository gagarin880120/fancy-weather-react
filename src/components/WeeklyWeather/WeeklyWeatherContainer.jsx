import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import WeeklyWeather from './WeeklyWeather';
import { selectWeeklyWeather } from '../../redux/selectors';

export function WeeklyWeatherContainer({ weeklyWeather }) {
  console.log(weeklyWeather);
  return (
    weeklyWeather
      ? (
        <WeeklyWeather
          weeklyWeather={weeklyWeather}
        />
      ) : <div>Loading...</div>
  );
}

export const mapStateToProps = (state) => ({
  weeklyWeather: selectWeeklyWeather(state),
});

WeeklyWeatherContainer.propTypes = {
  weeklyWeather: PropTypes.instanceOf(Array),
};

WeeklyWeatherContainer.defaultProps = {
  weeklyWeather: [],
};

export default connect(mapStateToProps)(WeeklyWeatherContainer);
