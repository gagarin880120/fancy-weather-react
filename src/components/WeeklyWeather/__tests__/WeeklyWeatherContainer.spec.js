import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { WeeklyWeatherContainer, mapStateToProps, mapDispatchToProps } from '../WeeklyWeatherContainer';

describe('WeeklyWeatherContainer', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<WeeklyWeatherContainer />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    test('should return right value', () => {
      const initialState = {
        language: 'en',
        temperatureScale: 'celsius',
        weeklyWeather: [
          {
            datetime: '2020-07-21',
            temp: 19.9,
            weather: {
              code: 804,
              icon: 'c04d',
            }
          },
          {
            datetime: '2020-07-22',
            temp: 15.6,
            weather: {
              code: 804,
              icon: 'c04d',
            }
          },
          {
            datetime: '2020-07-23',
            temp: 15.3,
            weather: {
              code: 804,
              icon: 'c04d',
            }
          }
        ],
      }

      expect(mapStateToProps(initialState)).toEqual({
        weeklyWeather: [
          {
            datetime: '2020-07-21',
            weekday: 'Tuesday',
            temp: 20,
            icon: 'DaySunnyOvercast',
            weather: {
              code: 804,
              icon: 'c04d',
            }
          },
          {
            datetime: '2020-07-22',
            weekday: 'Wednesday',
            temp: 16,
            icon: 'DaySunnyOvercast',
            weather: {
              code: 804,
              icon: 'c04d',
            }
          },
          {
            datetime: '2020-07-23',
            weekday: 'Thursday',
            temp: 15,
            icon: 'DaySunnyOvercast',
            weather: {
              code: 804,
              icon: 'c04d',
            }
          }
        ],
      });
    });
  });
});
