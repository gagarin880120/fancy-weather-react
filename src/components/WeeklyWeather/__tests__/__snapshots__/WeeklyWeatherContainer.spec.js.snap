// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyWeatherContainer component should render without crashing 1`] = `
<WeeklyWeather
  weeklyWeather={Array []}
/>
`;
